src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,393 @@
+/* eslint-disable es-x/no-function-prototype-bind -- Polyfill imported */
+
+import { normaliseDataset } from '../../common/normalise-dataset.mjs'
+import { nodeListForEach, mergeConfigs, extractConfigByNamespace } from '../../common.mjs'
+import { I18n } from '../../i18n.mjs'
+import '../../vendor/polyfills/Element/prototype/classList.mjs'
+import '../../vendor/polyfills/Element/prototype/dataset.mjs'
+import '../../vendor/polyfills/Function/prototype/bind.mjs'
+
+/**
+ * @constant
+ * @type {ExitThisPageTranslations}
+ * @see Default value for {@link ExitThisPageConfig.i18n}
+ * @default
+ */
+var EXIT_THIS_PAGE_TRANSLATIONS = {
+  activated: 'Loading.',
+  timedOut: 'Exit this page expired.',
+  pressTwoMoreTimes: 'Shift, press 2 more times to exit.',
+  pressOneMoreTime: 'Shift, press 1 more time to exit.'
+}
+
+/**
+ * Exit This Page component
+ *
+ * @class
+ * @param {HTMLElement} $module - HTML element that wraps the Exit This Page button
+ * @param {ExitThisPageConfig} [config] - Exit This Page config
+ */
+function ExitThisPage ($module, config) {
+  /** @type {ExitThisPageConfig} */
+  var defaultConfig = {
+    i18n: EXIT_THIS_PAGE_TRANSLATIONS
+  }
+
+  /**
+   * @deprecated Will be made private in v5.0
+   * @type {ExitThisPageConfig}
+   */
+  this.config = mergeConfigs(
+    defaultConfig,
+    config || {},
+    normaliseDataset($module.dataset)
+  )
+
+  this.i18n = new I18n(extractConfigByNamespace(this.config, 'i18n'))
+
+  /** @deprecated Will be made private in v5.0 */
+  this.$module = $module
+
+  /** @deprecated Will be made private in v5.0 */
+  this.$button = $module.querySelector('.govuk-exit-this-page__button')
```

Comment:
We redirect to the button's `href` a bit later on:

```js
window.location.href = this.$button.getAttribute('href')
```

Think we should check `instanceof HTMLAnchorElement` then can safely access `.href` instead

Reply:
I wonder if I should check this earlier on in the constructor function? We currently don't do any `$module` verification like we do for other components. I could check `$module` and `$button` at least at the outset?

Reply:
Sounds brill. Yeah try and check them both before adding to `this.$xxx`

Then the compiler knows exactly what they are later 👍